src/main/java/org/ambraproject/wombat/service/SearchFilterService.java

```diff
@@ -73,87 +74,36 @@
-
-    ArticleSearchQuery.Builder subjectAreaFacetQuery = ArticleSearchQuery.builder()
-        .setFacet(SUBJECT_AREA_FACET_FIELD)
-        .setQuery(query.getQuery().orElse(null))
-        .setSimple(query.isSimple())
-        .setArticleTypes(query.getArticleTypes())
-        .setAuthors(query.getAuthors())
-        .setDateRange(query.getDateRange().orElse(null))
-        .setJournalKeys(query.getJournalKeys())
-        .setSections(query.getSections())
-        .setSubjects(query.getSubjects());  // pass the previously filtered subjects to narrow the results
-
-    Map<?, ?> subjectAreaFacetResults = solrSearchApi.search(subjectAreaFacetQuery.build(), site);
-    SearchFilter subjectAreaFilter = searchFilterFactory
-        .createSearchFilter(subjectAreaFacetResults, SUBJECT_AREA, urlParams);
-
-    ArticleSearchQuery.Builder authorFacetQuery = ArticleSearchQuery.builder()
-        .setFacet(AUTHOR_FACET)
-        .setQuery(query.getQuery().orElse(null))
-        .setSimple(query.isSimple())
-        .setJournalKeys(query.getJournalKeys())
-        .setArticleTypes(query.getArticleTypes())
-        .setDateRange(query.getDateRange().orElse(null))
-        .setAuthors(query.getAuthors()) // pass the previously filtered authors to narrow the results
-        .setSubjects(query.getSubjects())
-        .setSections(query.getSections());
-
-
-    Map<?, ?> authorFacetResults = solrSearchApi.search(authorFacetQuery.build(), site);
-    SearchFilter authorFilter = searchFilterFactory.createSearchFilter(authorFacetResults, AUTHOR, urlParams);
-
-    ArticleSearchQuery.Builder articleTypeFacetQuery = ArticleSearchQuery.builder()
-        .setFacet(ARTICLE_TYPE_FACET)
-        .setQuery(query.getQuery().orElse(null))
-        .setSimple(query.isSimple())
-        .setDateRange(query.getDateRange().orElse(null))
-        .setJournalKeys(query.getJournalKeys())
-        .setSubjects(query.getSubjects())
-        .setAuthors(query.getAuthors())
-        .setSections(query.getSections());
-
-    Map<?, ?> articleTypeFacetResults = solrSearchApi.search(articleTypeFacetQuery.build(), site);
-    SearchFilter articleTypeFilter = searchFilterFactory.createSearchFilter(articleTypeFacetResults,
-        ARTICLE_TYPE, urlParams);
-
-    ArticleSearchQuery.Builder sectionFacetQuery = ArticleSearchQuery.builder()
-        .setFacet(SECTION_FACET)
-        .setIsPartialSearch(true)
-        .setQuery(query.getQuery().orElse(null))
-        .setSimple(query.isSimple())
-        .setDateRange(query.getDateRange().orElse(null))
-        .setJournalKeys(query.getJournalKeys())
-        .setSubjects(query.getSubjects())
-        .setAuthors(query.getAuthors());
-
-    Map<?, ?> sectionFacetResults = solrSearchApi.search(sectionFacetQuery.build(), site);
+  public Map<String, SearchFilter> getSearchFilters(ArticleSearchQuery query) throws IOException {
+    ImmutableList<String> facetFields = ImmutableList.of(JOURNAL_FACET_FIELD,
```

Comment:
Just noting that the inconsistent use of the `_FIELD` suffix annoys me.

Reply:
fixed in 5e795c06d